csp-tech-radar.csv

```diff
@@ -19,13 +19,14 @@ Anemic REST,hold,techniques,TRUE,"Failure to properly model the domain as a set
+ActiveMQ,adopt,platforms,TRUE,"Apache ActiveMQ is an open source message broker written in Java together with a full Java Message Service (JMS) client. KahaDB is the default storage database. Still used in a number of projects, with intentions to migrate towards Kafka. ActiveMQ is a push based system, whereas Kafka is pull based system with a more efficient storage format. Please contact the CSP SRE team for more information."
 Couchbase,adopt,platforms,TRUE,"Couchbase (not to be confused with CouchDB) is a fast, scalable NoSQL database that is well used and trusted within Sky OTT. Its masterless architecture makes it easier to scale and maintain than MongoDB and its built-in caching layer gives it better performance. Depending on your requirements, Cassandra is still top when it comes to Availability and Fault Tolerance, however Couchbase’s default CP configuration makes it a good option for ensuring Consistency."
 MySQL,adopt,platforms,TRUE,"MySQL is a fast, reliable and stable open source database management system. It has most of the features which oracle provides.  It is a relational database and is suitable where persisting data in a relational way is beneficial. It is being used by PD/PMP."
 .Net Core,trial,platforms,TRUE,".NET Core is an open source modular platform for creating applications that can be easily deployed in Windows, macOS and Linux. .NET Core makes it possible to build cross-platform web applications using ASP.NET Core with a set of tools, libraries and frameworks—another choice for microservices architecture."
 CockroachDB,assess,platforms,TRUE,"CockroachDB is a very promising new database technology based on Google’s Spanner database. It allows developers to run and scale it like any stateless service in Kubernetes while offering consistent ACID transactions and good performance. On top that it offers a SQL compliant interface. Almost sounds too good to be true, which is why we would like to assess this for now. An interesting article on running it in k8s can be found <a href=""https://dzone.com/articles/kubernetes-distributed-stateful-apps-using-cockroa"">here</a>."
 Windows Containers,assess,platforms,FALSE,"To achieve legacy in a box for many of our CSP applications that target the .Net Framework, Windows Containers should be explored as a possible solution."
 Oracle,hold,platforms,TRUE,Proprietary software is discouraged by the Open Source Software culture.
+Hazelcast,assess,platforms,FALSE,"For database caching, Hazelcast IMDG stores frequently accessed data in memory across an elastically scalable data grid. With its write-through and write-behind capabilities, Hazelcast can propagate any changes in the cached data back to the original store either synchronously or asynchronously. Has been adopted in a few teams at Sky."
```

Comment:
Please can we add something about the EntryProcessor functionality to allow state transformations to be done on the node holding the data in a single thread. This helps avoid concurrency issues.

Reply:
I've updated the Hazelcast description and pushed the changes to this branch.